src/test/chain-simulator/events.cs-e2e.ts

```diff
@@ -0,0 +1,105 @@
+import { config } from "./config/env.config";
+import { ChainSimulatorUtils } from "./utils/test.utils";
+import axios from "axios";
+
+describe('Events e2e tests with chain simulator', () => {
+  beforeAll(async () => {
+    await ChainSimulatorUtils.waitForEpoch(2);
+  });
+
+  describe('GET /events', () => {
+    it('should return status code 200 and a list of events', async () => {
+      const response = await axios.get(`${config.apiServiceUrl}/events`);
+      expect(response.status).toBe(200);
+      expect(response.data).toBeInstanceOf(Array);
+    });
+
+    it('should support pagination', async () => {
+      const response = await axios.get(`${config.apiServiceUrl}/events?from=0&size=2`);
+      expect(response.status).toBe(200);
+      expect(response.data).toBeInstanceOf(Array);
+      expect(response.data.length).toStrictEqual(2);
+    });
+
+    it('should return events with expected properties', async () => {
+      const response = await axios.get(`${config.apiServiceUrl}/events`);
+      const event = response.data[0];
+      expect(response.status).toBe(200);
+
+      const requiredProps = [
+        'txHash',
+        'logAddress',
+        'identifier',
+        'address',
+        'topics',
+        'shardID',
+        'txOrder',
+        'order',
+        'timestamp',
+      ];
+
+      for (const prop of requiredProps) {
+        expect(event).toHaveProperty(prop);
+      }
+
+      expect(event.topics).toBeInstanceOf(Array);
+    });
+
+    it('should support filtering by address', async () => {
+      const response = await axios.get(`${config.apiServiceUrl}/events?address=${config.aliceAddress}`);
+      expect(response.status).toBe(200);
+      expect(response.data).toBeInstanceOf(Array);
+      expect(response.data.length).toBeGreaterThan(1);
```

Comment:
maybe check that the events have the provided address

Reply:
done